### PR TITLE
Update CodeQL docs directory structure

### PR DESCRIPTION
**Pull request by jf205**
The PR revises the CodeQL documentation project structure. One test keeps failing, but I think the content changes are ready for review.

### CODE REVIEW

**Comment by adityasharad** [comment]
Tests are passing on the corresponding internal PR. I'm happy to merge this once it's been reviewed by someone else on the docs team, since it appears to be entirely content changes.

File: docs/codeql/codeql-language-guides/codeql-for-java/types-in-java.rst [hunk 1]
```diff
@@ -1,3 +1,5 @@
+.. _types-in-java:
+
 Java types
```
Comment (felicitymay) [inline docs/codeql/codeql-language-guides/codeql-for-java/types-in-java.rst:3]: Mismatch between map/filename and article title.

```suggestion
Types in Java
```

File: docs/codeql/codeql-language-guides/codeql-for-java/index.rst [hunk 2]
```diff
@@ -15,7 +17,7 @@ Experiment and learn how to write effective and efficient queries for CodeQL dat
    annotations-in-java
    javadoc
    working-with-source-locations
-   abstract-syntax-tree-classes-for-working-with-go-programs
+   abstract-syntax-tree-classes-for-working-with-java-programs
```
Comment (felicitymay) [inline docs/codeql/codeql-language-guides/codeql-for-java/index.rst:20]: Line 28 needs to change:

`Java types` -> `Types in Java`

Possibly there are other links to this article that will need their names updating.

File: docs/codeql/ql-language-reference/ql-language-specification.rst [hunk 3]
```diff
@@ -276,7 +277,7 @@ Type references
 
 With the exception of class domain types and character types (which cannot be referenced explicitly in QL source), a reference to a type is written as the name of the type. In the case of database types, the name includes the at sign (``@``, U+0040).
 
-::
+.. code-block:: ql
 
    type ::= (moduleId "::")? classname | dbasetype | "boolean" | "date" | "float" | "int" | "string"
```
Comment (felicitymay) [inline docs/codeql/ql-language-reference/ql-language-specification.rst:282]: The `?` symbol on line 282 (above) shows up in the preview with a red box round it. I have no idea why and whether or not this is a problem, but thought I should mention it.
image

Comment (felicitymay) [reply docs/codeql/ql-language-reference/ql-language-specification.rst:282]: This is also seen at other locations in this file. I'm suspicious that none of the code in this file should have QL syntax highlighting.

File: docs/codeql/ql-language-reference/qldoc-comment-specification.rst [hunk 4]
```diff
@@ -1,3 +1,5 @@
+.. _qldoc-comment-specification:
+
```
Comment (felicitymay) [inline docs/codeql/ql-language-reference/qldoc-comment-specification.rst:2]: Line 52 has a couple of `@` characters in single quotes that look as if they ought really to be in double-backticks. However, this is how they are in the current version.

**Review by felicitymay — DISMISSED** [review]
It is so great to see this falling into place following all the work you've put into it ✨ 💎 ✨ 

I followed your instructions for reviewing and focused on the preview. I found very few glitches, mostly it looks spot on. I flagged anything that occurred to me or that I spotted, so some of them are comments for the future.

I put some more general observations into the tracking issue.
Now off to look at the related internal PR.

File: docs/codeql/codeql-cli/using-the-codeql-cli/testing-custom-queries.rst [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+
+      **Information for LGTM Enterprise 1.23 users**
+
+   CodeQL tests are a new feature in CodeQL CLI version 2.0.2.
+   
+   If you are an LGTM Enterprise 1.23 user, you should still use CodeQL CLI
+   version 2.0.1 to prepare databases to upload to your instance of LGTM. You
+   can use version 2.0.2 (or newer) to test your custom queries, but databases
+   created with versions newer than 2.0.1 are not compatible with LGTM
+   Enterprise 1.23. For more information, see ":ref:`Getting started with the CodeQL CLI <using-two-versions-of-the-codeql-cli>`."
```
[the remark this answers is not in the document]
Comment (jf205) [reply docs/codeql/codeql-cli/using-the-codeql-cli/testing-custom-queries.rst:28]: 👍 https://github.com/github/semmle-docs/issues/269

File: docs/codeql/codeql-cli/using-the-codeql-cli/upgrading-codeql-databases.rst [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+CodeQL databases are upgraded by running the following command::
+
+   codeql database upgrade <database>
+
+where ``<database>``, the path to the CodeQL database you
+want to upgrade, must be specified.
+
+For full details of all the options you can use when upgrading databases,
+see the `database upgrade reference documentation <../codeql-cli-manual/database-upgrade.html>`.
```
[the remark this answers is not in the document]
Comment (jf205) [reply docs/codeql/codeql-cli/using-the-codeql-cli/upgrading-codeql-databases.rst]: The CLI manual is in a different Sphinx project, so we can't use :ref: links unfortunately. But there is also a typo here 🤦

File: docs/codeql/ql-language-reference/qldoc-comment-specification.rst (hunk 4 above)
Comment (jf205) [reply docs/codeql/ql-language-reference/qldoc-comment-specification.rst:2]: I'll change those to backticks 👍

File: docs/codeql/ql-language-reference/ql-language-specification.rst (hunk 3 above)
Comment (jf205) [reply docs/codeql/ql-language-reference/ql-language-specification.rst:282]: You're right. I'll turn off the syntax highlighting for this file.

**Review by jf205 — COMMENTED** [review]
Thanks for the thorough review and great suggestions @felicitymay ❤️

**Review by felicitymay — APPROVED** [review]
Thanks for the updates and new issue 😄 

:shipit:

**Comment by jf205** [comment]
> Tests are passing on the corresponding internal PR. I'm happy to merge this once it's been reviewed by someone else on the docs team, since it appears to be entirely content changes.

I don't seem to have permission to merge this, so I'll hand over to @adityasharad! (Perhaps you could also press the button on https://github.com/github/semmle-code/pull/38179 as well please.)

**Pull request merged**